packages/core/src/root.ts

```diff
@@ -273,6 +290,7 @@ export function createRemoteRoot<
   // Might need to send the removed child ID, or find out if we
   // can collect removals into a single update.
   function removeChild(container: HasChildren, child: CanBeChild) {
+    componentsSet.delete(child);
```

Comment:
You shouldn't remove it, there isn't technically anything wrong in removing a child, then re-appending that child to the tree somewhere else